src/theory/arrays/theory_arrays_rewriter.cpp

```diff
@@ -45,6 +45,622 @@ void setMostFrequentValueCount(TNode store, uint64_t count) {
   return store.setAttribute(ArrayConstantMostFrequentValueCountAttr(), count);
 }
 
+Node TheoryArraysRewriter::normalizeConstant(TNode node)
+{
+  return normalizeConstant(node, node[1].getType().getCardinality());
+}
+
+// this function is called by printers when using the option "--model-u-dt-enum"
+Node TheoryArraysRewriter::normalizeConstant(TNode node, Cardinality indexCard)
```

Comment:
I'm wondering why we need this normalization, but that is probably a discussion for some other time.